.github/workflows/build-full-application.yaml

```diff
@@ -0,0 +1,99 @@
+name: Build Full Application workflow
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: "The name of the environment to build into"
+        required: true
+        type: string
+      workspace:
+        description: "The name of the workspace to build the application into"
+        required: true
+        type: string
+      services:
+        description: "A list of the services to build"
+        required: false
+        default: ""
+        type: string
+      tag:
+        description: "The tag to build from"
+        required: false
+        default: ""
+        type: string
+      multi_domain:
+        description: "Domain to use for github runner role"
+        required: false
+        default: ""
+        type: string
+      workflow_timeout:
+        description: "Timeout duration in minutes"
+        required: false
+        default: 5
+        type: number
+      target_repo:
+        description: "The name of the repo"
+        required: true
+        type: string
+      target_repo_commit_hash:
+        description: "The name of the target repo branch"
+        required: true
+        type: string
+
+jobs:
+  deploy-infrastructure-apply:
```

Comment:
As discussed this may be for a later ticket but i think we need to deploy infrastructure from task branch of triggering repo and from main for others ; i don't think the code currently will handle this

Reply:
ticket to be created

Reply:
DR-1026